samples/dotnet/KernelHttpServer/SemanticKernelEndpoint.cs

```diff
@@ -87,50 +122,45 @@ public SemanticKernelEndpoint(IMemoryStore memoryStore)
 
         var kernel = SemanticKernelFactory.CreateForRequest(
             req,
-            executionContext.GetLogger<SemanticKernelEndpoint>());
+            executionContext.GetLogger<SemanticKernelEndpoint>(),
+            ask.Skills);
 
         if (kernel == null)
         {
             return await req.CreateResponseWithMessageAsync(HttpStatusCode.BadRequest, "Missing one or more expected HTTP Headers");
         }
 
-        var contextVariables = new ContextVariables(ask.Value);
-
-        foreach (var input in ask.Inputs)
-        {
-            contextVariables.Set(input.Key, input.Value);
-        }
+        var context = kernel.CreateNewContext();
 
-        var planner = kernel.Skills.GetFunction("plannerskill", "executeplan");
-        var result = await kernel.RunAsync(contextVariables, planner);
+        var plan = Plan.FromJson(ask.Value, context);
```

Comment:
Do you really need a context when loading from Json? Supposedly the Context would come also from the Json right?

Reply:
Json will have ContextVariables -- but not an SKContext. Needed as this will have the `Skills` to load functions from. If not registered, plan deserialization will fail.